File: Recursion/4.3-Maximum-Number-Using-Recursion/c/code.c
#include <stdio.h>

int max(int arr[], int idx, int length)
{

    if (idx == length - 1)
    {
        return arr[idx - 1];
    }

    int current = arr[idx];

    int other = max(arr, idx + 1, length);

    int result = (current > other) ? current : other;

    return result;
}

int main()
{

    int arr[] = {1, 10, 2, 3, 2, 9};

    int length = sizeof(arr) / sizeof(arr[0]);
    int maximum = max(arr, 0, length);

    printf("Maximum: %d\n", maximum);

    return 0;
}

File: Recursion/4.3-Maximum-Number-Using-Recursion/go/code.go
package main

func maximum(arr []int) int {
	if len(arr) == 1 {
		return arr[0]
	}

	var current int = arr[0]

	var other int = maximum(arr[1:])

	if current < other {
		return other
	}

	return current

}

func main() {
	arr := []int{1, 2, 6, 4, 5}
	var max int = maximum(arr)

	println(max)
}
